File: JavaScript/leetcode75/src/main/Problem10_MoveZeroes.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */
var moveZeroes = function (nums) {
  let index = 0;
  for (let i = 0; i < nums.length; i++) {
    if (nums[i] != 0) {
      let temp = nums[i];
      nums[i] = nums[index];
      nums[index++] = temp;
    }
  }
  return nums;
};

module.exports = moveZeroes;

File: Java/leetcode75/src/main/java/org/phoenix/leetcode/challenges/Problem10_MoveZeroes.java
package org.phoenix.leetcode.challenges;

public class Problem10_MoveZeroes {
    public int[] moveZeroes(int[] nums) {
        int index = 0;
        for (int i = 0; i < nums.length; i++) {
            if (nums[i] != 0) {
                int temp = nums[i];
                nums[i] = nums[index];
                nums[index++] = temp;
            }
        }
        return nums;
    }
}
